client/components/Main.jsx

```diff
@@ -6,21 +6,33 @@ import Navbar from '../containers/NavbarContainer';
 import Home from './Home';
 import Products from '../containers/ProductsContainer';
 import Login from '../containers/AuthContainer';
-import Cart from './Cart';
+import Cart from '../containers/CartContainer';
 import SingleProduct from '../containers/SingleProductContainer';
 import '../stylesheets/index.scss';
 import '../stylesheets/header.scss';
 
 const propTypes = {
   getProducts: PropTypes.func.isRequired,
   onLoad: PropTypes.func.isRequired,
+  getCartByUser: PropTypes.func.isRequired,
+  getCartBySession: PropTypes.func.isRequired,
+  signedIn: PropTypes.bool.isRequired,
+  // eslint-disable-next-line react/require-default-props
+  userId: PropTypes.number,
+  sessionId: PropTypes.number.isRequired,
 };
 
 export default class Main extends Component {
   componentDidMount() {
-    const { getProducts, onLoad } = this.props;
-    onLoad();
-    getProducts();
+    const { getProducts, onLoad, getCartByUser, getCartBySession, signedIn, userId, sessionId } = this.props;
+
+    if (onLoad() && getProducts()) {
```

Comment:
I'm not sure if we need this, I think we could take advantage of either middleware or existing action creators to get a session/user's order